airflow/sensors/external_task.py

```diff
@@ -351,29 +348,30 @@ def execute(self, context: Context) -> None:
             super().execute(context)
         else:
             self.defer(
-                trigger=TaskStateTrigger(
-                    dag_id=self.external_dag_id,
-                    task_id=self.external_task_id,
+                timeout=self.execution_timeout,
+                trigger=WorkflowTrigger(
+                    external_dag_id=self.external_dag_id,
+                    external_task_ids=self.external_task_ids,
                     execution_dates=self._get_dttm_filter(context),
-                    states=self.allowed_states,
-                    trigger_start_time=utcnow(),
-                    poll_interval=self.poll_interval,
+                    allowed_states=self.allowed_states,
+                    poke_interval=self.poll_interval,
+                    soft_fail=self.soft_fail,
                 ),
                 method_name="execute_complete",
             )
 
     def execute_complete(self, context, event=None):
         """Execute when the trigger fires - return immediately."""
         if event["status"] == "success":
-            self.log.info("External task %s has executed successfully.", self.external_task_id)
-            return None
-        elif event["status"] == "timeout":
-            raise AirflowException("Dag was not started within 1 minute, assuming fail.")
+            self.log.info("External tasks %s has executed successfully.", self.external_task_ids)
         else:
-            raise AirflowException(
-                "Error occurred while trying to retrieve task status. Please, check the "
-                "name of executed task and Dag."
-            )
+            if self.soft_fail:
```

Comment:
https://github.com/apache/airflow/blob/c8775fc0122fdb7243728c76cde7368509edf725/airflow/sensors/base.py#L301-L304

I think we might not need to catch this for core sensors.

Reply:
If an exception occurs in the trigger code then this base implementation takes care of soft fail. I have this check if handle if trigger has any event without success then skip is it if soft_fail set instead of fail

Reply:
got it. make sense